src/share/classes/com/sun/btrace/agent/TraceOutputWriter.java

```diff
@@ -149,7 +149,7 @@ private void nextWriter() {
         private FileWriter getNextWriter() throws IOException {
         	currentFileWriter.close();
         	File scriptOutputFile_renameFrom = new File(path + File.separator + baseName);
-        	File scriptOutputFile_renameTo = new File(path + File.separator + baseName + "." + (counter++));
+        	File scriptOutputFile_renameTo = new File(path + File.separator + baseName.substring(0, baseName.indexOf("-")) + ".btrace." + (counter++));
```

Comment:
I agree. This is particularly bad form. However, I end up with output files named "TraceMem.class-.MontricAgent.1461952385969.btrace" which seems to have a redundant dash in them. Maybe it would be better to find and remove this one instead. 

I will investigate a bit on this issue.